### PR TITLE
Katie's breakfast scrabble game

### PR DESCRIPTION
**Pull request by ktrops**
This code is working, but I could do a lot more to clean up the methods. I just ran out of time.

### CODE REVIEW

File: lib/scrabble.rb [hunk 1]
```diff
@@ -1,4 +1,63 @@
 module Scrabble
   # require your gems and classes here
   # require_relative 'scrabble/whatevs'
+  class Scrabble
+
+    def self.score(word)
+      value = {
```
Comment (kariabancroft) [inline lib/scrabble.rb:7]: This would be a great use of a constant - also a more descriptive variable name since later with minimal context you wouldn't really know what value is referring to

File: lib/scrabble.rb [hunk 2]
```diff
@@ -1,4 +1,63 @@
...
+                 5 => ["k"],
+                 8 => ["j", "x"],
+                 10 => ["q", "z"]
+                }
+
+      count = 0
+        word_array = word.to_s.downcase.split(//)
+        return "too many letters" if word_array.length > 7 
+        return "error" if word.to_s.strip.empty?
+        word_array.each do |l|
```
Comment (kariabancroft) [inline lib/scrabble.rb:21]: I would prefer if you used a more descriptive variable name like `letter`, `key` and `value` (other something else since you're already using `value`

File: lib/scrabble.rb [hunk 3]
```diff
@@ -1,4 +1,63 @@
...
+      high_score = 0
+      array_of_words.each_with_index do |word, index|
+        score = self.score(word)
+        if score > high_score
+          winner_word.replace([word])
+          high_score = score
+        elsif score == high_score
+          # if winner_word[0].length == 7
+          #   return winner_word[0]
+          if winner_word[0].length == word.length
```
Comment (kariabancroft) [inline lib/scrabble.rb:46]: This if statement has me a bit confused - do you actually need to use the `winner_word` array if you are normally just using the first element in the array? Is there a scenario where this array will have more than one value?

File: spec/scrabble_spec.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
+require './lib/scrabble.rb'
+require 'spec_helper.rb'
+
+describe "Scrabble" do
+  # include Scrabble
+  context "#self.score" do
```
Comment (kariabancroft) [inline spec/scrabble_spec.rb:6]: I would try to organize your error scenarios separate from your "success" scenarios (so moving the first one to go along with the scenario on line 36, for ex)

File: spec/scrabble_spec.rb [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    it "returns the the word with 7 letters if score is equal" do
+      expect(Scrabble::Scrabble.highest_score_from(["ANGRILY", "qa"])).to eq ("ANGRILY")
+    end
+
+    it "returns the first word in list if length and score is the same" do
+      expect(Scrabble::Scrabble.highest_score_from(["angrily", "deadbul", "dog", "hello"])).to eq ("angrily")
+    end
+  end
+
```
Comment (kariabancroft) [inline spec/scrabble_spec.rb:58]: Remove your extra whitespace

File: spec/scrabble_spec.rb [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    it "returns the highest scoring word" do
+      expect(Scrabble::Scrabble.highest_score_from(["apple", "hello"])).to eq ("apple")
+    end
+
+    it "returns the shortest word if score is equal" do
+      expect(Scrabble::Scrabble.highest_score_from(["dog", "long"])).to eq ("dog")
+    end
+
+    it "returns the the word with 7 letters if score is equal" do
+      expect(Scrabble::Scrabble.highest_score_from(["ANGRILY", "qa"])).to eq ("ANGRILY")
```
Comment (kariabancroft) [inline spec/scrabble_spec.rb:51]: I appreciate that you had an edge case in here too - having a fully capitalized word as well

**Comment by kariabancroft** [comment]
Great job! Your whitespace is consistent and your tests are great. I'm a bit confused about your use of an array for the winning word - possibly some comments would help clarify. Try to use a bit more whitespace between conditionals in the future to ensure that there is logical separation between big blocks. I really liked the way you used some of those one-line `if` statements!
